packages/gatsby/src/commands/serve.js

```diff
@@ -87,6 +149,8 @@ module.exports = async program => {
   router.use(compression())
   router.use(express.static(`public`))
   const matchPaths = await readMatchPaths(program)
+  const truncatedPaths = await readTruncatedPaths(program)
+  router.use(truncatedPathRouter(truncatedPaths, { root }))
```

Comment:
I know this is marked as WIP, but changing `gatsby serve` router is a serious red flag (as it implies additional configuration needed for hosting your sites in production)

Reply:
If this is needed, there probably be explicit opt-in into using this stuff (+ error when not using it and exceeding the character limit)

Reply:
Should this rather be something like a flag to `serve` ?

Reply:
The thing is that `gatsby build` is producing some different artificats if character limit is exceeded so this not only affect local builds and `gatsby serve` but also production sites build in CIs etc - so this is dangerous to do this by default (because hosting for production site would need to be configured to support that)

Reply:
I see. That's indeed dangerous. So the simplest solution is to only throw a detailed error when names are too long. Additionally, something like `gatsby-config` could be used to opt into automatic truncating. Should I make error-only for now?